periodic-kueue-verify-website-links-release-0-18 failed or missing link

<!--
Please use this template for reporting flaky tests.
Links to specific failures in Prow are appreciated.
-->

**Which test is flaking?**:
periodic-kueue-verify-website-links-release-0-18 failed or missing link

**Link to failed CI job or steps to reproduce locally**:

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/periodic-kueue-verify-website-links-release-0-18/2066179674119081984
**Failure message or logs**:
```
10 threads active,   196 links queued, 1862 links in 2068 URLs checked, runtime 2 minutes, 6 seconds
10 threads active,   182 links queued, 1897 links in 2089 URLs checked, runtime 2 minutes, 11 seconds
10 threads active,   170 links queued, 1909 links in 2090 URLs checked, runtime 2 minutes, 16 seconds
10 threads active,   159 links queued, 1920 links in 2090 URLs checked, runtime 2 minutes, 21 seconds
10 threads active,   145 links queued, 1934 links in 2090 URLs checked, runtime 2 minutes, 26 seconds
10 threads active,   131 links queued, 1948 links in 2090 URLs checked, runtime 2 minutes, 31 seconds
10 threads active,   117 links queued, 1962 links in 2090 URLs checked, runtime 2 minutes, 36 seconds
URL        `/examples/jobs/ray-service-sample.yaml'
Name       `ray-service-sample.yaml'
Parent URL https://kueue.sigs.k8s.io/docs/tasks/run/rayservices/, line 213, col 458
Real URL   https://kueue.sigs.k8s.io/examples/jobs/ray-service-sample.yaml
Check time 2.778 seconds
Result     Error: ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
10 threads active,   102 links queued, 1977 links in 2090 URLs checked, runtime 2 minutes, 41 seconds
10 threads active,    88 links queued, 1991 links in 2090 URLs checked, runtime 2 minutes, 46 seconds
10 threads active,    75 links queued, 2004 links in 2090 URLs checked, runtime 2 minutes, 51 seconds
10 threads active,    59 links queued, 2020 links in 2095 URLs checked, runtime 2 minutes, 56 seconds
10 threads active,    45 links queued, 2034 links in 2099 URLs checked, runtime 3 minutes, 1 seconds
10 threads active,    32 links queued, 2047 links in 2102 URLs checked, runtime 3 minutes, 6 seconds
10 threads active,    16 links queued, 2063 links in 2108 URLs checked, runtime 3 minutes, 11 seconds
10 threads active,     2 links queued, 2077 links in 2108 URLs checked, runtime 3 minutes, 16 seconds
Statistics:
Downloaded: 50.5MB.
Content types: 18 image, 403 text, 0 video, 0 audio, 142 application, 0 mail and 1526 other.
URL lengths: min=15, max=1068, avg=123.
That's it. 2089 links in 2108 URLs checked. 0 warnings found (58 ignored or duplicates not printed). 1 error found.
Stopped checking at 2026-06-14 15:27:11+000 (3 minutes, 21 seconds)
make: *** [Makefile-verify.mk:272: verify-website-links] Error 1
```

**Anything else we need to know?**:

--- a/hack/testing/linkchecker/verify.sh
+++ b/hack/testing/linkchecker/verify.sh
@@ -19,12 +19,14 @@ set -o nounset
 set -o pipefail
 
 DOCKER="${DOCKER:-docker}"
-CURRENT_DIR=$(dirname "${BASH_SOURCE[0]}")
+SOURCE_DIR="$(cd "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)"
+ROOT_DIR="${SOURCE_DIR}/../../.."
 
 echo "Building linkchecker Docker image..."
-"${DOCKER}" build --load -f "${CURRENT_DIR}/Dockerfile" -t linkchecker "${CURRENT_DIR}"
+"${DOCKER}" build --load -f "${SOURCE_DIR}/Dockerfile" -t linkchecker "${SOURCE_DIR}"
 
 echo "Running linkchecker..."
-"${DOCKER}" run --rm linkchecker --no-warnings --ignore-url='^mailto:' --ignore-url='^tel:' https://kueue.sigs.k8s.io/
+"${ROOT_DIR}/hack/testing/retry.sh" --attempts 5 --delay 5 --stream -- \
+    "${DOCKER}" run --rm linkchecker --no-warnings --ignore-url='^mailto:' --ignore-url='^tel:' https://kueue.sigs.k8s.io/
 
 echo "Link check completed successfully"

